### ISSUE TITLE
No C++ classes and methods?

### ISSUE THREAD

**Issue by patlecat** [intel/hyperscan#227]
When I look at the examples I see only the C interface being used. That's so weird since 80% of the lib was written in C++!?

Is there something for C++?

**Comment by xiangwang1** [comment]
No, but I don't think there could be large efforts using these APIs in C++.

Hyperscan compile time is written with C++ to make our analysis easier, but run-time is more performance critical and therefore is written in C. Ultimately we need to have C structures for run-time matching. 

Plus most of our target users are C based.

**Comment by patlecat** [comment]
I see, but I think that's only half the story and not enough argument against not just offering a proper C++ interface that performs well, I mean it's not like offering a .NET interface where performance goes out the window. This is C++ and it can be optimized very well with the most recent compilers!
